server/src/main/java/org/elasticsearch/search/aggregations/pipeline/AbstractPipelineAggregationBuilder.java

```diff
@@ -102,6 +105,31 @@ public final PipelineAggregator create() throws IOException {
     public void doValidate(AggregatorFactory<?> parent, Collection<AggregationBuilder> factories,
             Collection<PipelineAggregationBuilder> pipelineAggregatorFactories) {
     }
+    
+    public static void doValidateParentAggregations(AggregatorFactory<?> parent, String name) {
+        if (parent instanceof HistogramAggregatorFactory) {
+            HistogramAggregatorFactory histoParent = (HistogramAggregatorFactory) parent;
+            if (histoParent.minDocCount() != 0) {
+                throw new IllegalStateException("parent histogram of derivative aggregation [" + name
```

Comment:
Since this is shared by a bunch of aggs now, I think we need to tweak the exception message to use the `type` variable:

`"parent histogram of " + type + " aggregation..."`

Same for the other exception below.

I'm not positive, but this may affect some of the unit tests elsewhere (we sometimes check exception messages).  Just FYI some of the tests may need slight tweaking if the exception message changes.